cmd/ion-go/eventwriter.go

```diff
@@ -194,11 +195,13 @@ func (e *eventwriter) BeginStruct() error {
 		return err
 	}
 	e.depth++
+	e.inStruct = true
 	return nil
 }
 
 func (e *eventwriter) EndStruct() error {
 	e.depth--
+	e.inStruct = false
```

Comment:
`{structs:{can:"end"},inside:{other:"structs"}}` :(

Reply:
Good catch. I'll fix it.